src/Discord/Builders/Components/Thumbnail.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file is a part of the DiscordPHP project.
+ *
+ * Copyright (c) 2015-present David Cole <david.cole1340@gmail.com>
+ *
+ * This file is subject to the MIT license that is bundled
+ * with this source code in the LICENSE.md file.
+ */
+
+namespace Discord\Builders\Components;
+
+/**
+ * Thumbnail components allow you to add a thumbnail image to a section.
+ *
+ * @link https://discord.com/developers/docs/interactions/message-components#thumbnail
+ *
+ * @since 10.4.0
+ */
+class Thumbnail extends Component implements Contracts\ComponentV2
+{
+    /**
+     * Media item for the thumbnail.
+     *
+     * @var array
+     */
+    private $media;
+
+    /**
+     * Description for the thumbnail.
+     *
+     * @var string|null
+     */
+    private $description;
+
+    /**
+     * Whether the thumbnail is a spoiler.
+     *
+     * @var bool
+     */
+    private $spoiler = false;
+
+    /**
+     * Creates a new thumbnail.
+     *
+     * @param string $url URL of the media item.
+     *
+     * @return self
+     */
+    public static function new(string $url): self
+    {
+        $component = new self();
+        $component->setMedia($url);
+
+        return $component;
+    }
+
+    /**
+     * Sets the media item for the thumbnail.
+     *
+     * @param string $url URL of the media item.
+     *
+     * @return $this
+     */
+    public function setMedia(string $url): self
+    {
+        $this->media = ['url' => $url];
```

Comment:
The spec actually expects the media to contain UnfurledMediaItem, even though only the `url` for now that will be taken, that is what you can deal with inside jsonSerialize()

Needs to change on the getter too


For example

```suggestion
     * @param UnfurledMediaItem|string $url URL of the media item.
     *
     * @return $this
     */
    public function setMedia(UnfurledMediaItem|string $url): self
    {
        if (is_string($url)) {
            $url = UnfurledMediaItem::new($url);
        }

        $this->media = $url;
```

with this we can anticipate more changes in future